feature/settings/src/main/java/com/google/jetpackcamera/settings/SettingsScreen.kt

```diff
@@ -16,20 +16,273 @@
 
-import androidx.compose.material3.Text
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.selection.selectableGroup
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.*
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.google.jetpackcamera.settings.model.DarkModeStatus
+import com.google.jetpackcamera.settings.model.Settings
+
+
+private const val TAG = "SettingsScreen"
 
 /**
  * Screen used for the Settings feature.
  */
 @Composable
 fun SettingsScreen(
-    viewModel: SettingsViewModel = hiltViewModel()
-) {
+    viewModel: SettingsViewModel = hiltViewModel(),
+    navController: NavController) {
     val settingsUiState by viewModel.settingsUiState.collectAsState()
 
-    Text(settingsUiState.text)
-}
\ No newline at end of file
+    Column(
+        modifier = Modifier
+            .verticalScroll(rememberScrollState())
+    ) {
+        settingsPageHeader(
+            title = stringResource(id = R.string.settings_title),
+            navBack = {
+                navController.popBackStack()
+            }
+        )
+
+        SettingsList(uiState = settingsUiState, viewModel = viewModel)
+
+    }
+}
+
+@Composable
+fun SettingsList(uiState: SettingsUiState, viewModel: SettingsViewModel) {
+    sectionHeader(title = stringResource(id = R.string.section_title_camera_settings))
+
+    defaultCameraFacing(
+        settings = uiState.settings,
+        onClick = viewModel::setDefaultToFrontCamera
+    )
+
+    sectionHeader(title = stringResource(id = R.string.section_title_app_settings))
+    DarkModeSetting(
+        uiState = uiState,
+        setDarkMode = viewModel::setDarkMode)
+}
+
+/**
+ * MAJOR SETTING UI COMPONENTS
+ * these are ready to be popped into the ui
+ */
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun settingsPageHeader(title: String, navBack: () -> Unit) {
```

Comment:
Composable functions should start with an upper case letter. Also lets move these Composables outof the SettingsScreen.kt